### PR TITLE
Add flags for AMD GFX906 MI50/MI60

### PR DESCRIPTION
**Pull request by bjoo**
Hi. I've added hooks for the AMD GFX906 architecture for MI50/MI60.
Right now this just defines KOKKOS_ENABLE_AMD_GFX906 in KokkosCore_Config.h and appends --amdgpu-target=gfx906 to the link line only.

### CODE REVIEW

File: cmake/kokkos_arch.cmake [hunk 1]
```diff
@@ -222,6 +223,14 @@ IF (KOKKOS_ARCH_EPYC)
   )
 ENDIF()
 
+IF (KOKKOS_ARCH_GFX906)
+  SET(KOKKOS_ARCH_AMD_GFX906 ON)
+  COMPILER_SPECIFIC_LINKER_FLAGS(
+	LINK_OPTIONS
+        DEFAULT --amdgpu-target=gfx906
+  )
+ENDIF()
```
Comment (crtrott) [inline cmake/kokkos_arch.cmake]: We need to do this correct based on which backends are enabled and which compiler. This architecture will be relevant both for HIP and OpenMPTarget. I have a branch on my system to add some of this correctly.

**Comment by bjoo** [comment]
Agreed.

If you have a template for how you’d like it too look I can make it look like that :)

Best, 
 B

> On Feb 12, 2020, at 4:22 PM, Christian Trott <notifications@github.com> wrote:
> 
> @crtrott commented on this pull request.
> 
> In cmake/kokkos_arch.cmake:
> 
> > @@ -222,6 +223,14 @@ IF (KOKKOS_ARCH_EPYC)
>    )
>  ENDIF()
>  
> +IF (KOKKOS_ARCH_GFX906)
> +  SET(KOKKOS_ARCH_AMD_GFX906 ON)
> +  COMPILER_SPECIFIC_LINKER_FLAGS(
> +	LINK_OPTIONS
> +        DEFAULT --amdgpu-target=gfx906
> +  )
> +ENDIF() 
> 
> We need to do this correct based on which backends are enabled and which compiler. This architecture will be relevant both for HIP and OpenMPTarget. I have a branch on my system to add some of this correctly.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.
> 

-----------------------------------------------------------------------------
Dr Balint Joo   High Performance Computational Scientist
Jefferson Lab
12000 Jefferson Ave Suite 3, MS 12B2, Room F217
Newport News VA 23606, USA
Tel: +1-757-269-5339         email: bjoo AT jlab.org
-----------------------------------------------------------------------------

**Comment by crtrott** [comment]
The branch I was working on is on my fork: crtrott/openmptarget-amd

**Comment by crtrott** [comment]
superceded by #2792

**Pull request closed without merging**